src/test/java/com/ning/http/client/async/BasicProxyAuthorizationTest.java

```diff
@@ -0,0 +1,168 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.ning.http.client.async;
+
+import com.ning.http.client.AsyncHttpClient;
+import com.ning.http.client.AsyncHttpClientConfig;
+import com.ning.http.client.ProxyServer;
+import com.ning.http.client.Request;
+import com.ning.http.client.Response;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+
+import org.eclipse.jetty.http.HttpVersion;
+import org.eclipse.jetty.proxy.ConnectHandler;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.HttpConnectionFactory;
+import org.eclipse.jetty.server.SecureRequestCustomizer;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.SslConnectionFactory;
+import org.eclipse.jetty.server.handler.HandlerWrapper;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class BasicProxyAuthorizationTest extends AbstractBasicTest {
+
+    protected static final Logger log = LoggerFactory.getLogger(BasicProxyAuthorizationTest.class);
+    private static final String LOCALHOST = "127.0.0.1";
+    private static final String CONNECTION_HEADER = "X-Connection";
+    private static final String KEEP_ALIVE = "keep-alive";
+    private static final String PROXY_AUTHORIZATION_HEADER = "Proxy-Authorization";
+    private static final String NTLM_MSG_TYPE_1 = "NTLM TlRMTVNTUAABAAAAAYIIogAAAAAoAAAAAAAAACgAAAAFASgKAAAADw==";
+
+    @Override
+    public AsyncHttpClient getAsyncHttpClient(AsyncHttpClientConfig config) {
+        return ProviderUtil.grizzlyProvider(config);
+    }
+
+    public HandlerWrapper configureHandler() throws Exception {
+        return  new BasicProxyAuthorizationHandler();
+    }
+
+    public static class BasicProxyAuthorizationHandler extends ConnectHandler {
+
+        @Override
+        public void handle(String pathInContext, org.eclipse.jetty.server.Request request, HttpServletRequest httpRequest,
+                           HttpServletResponse httpResponse) throws IOException, ServletException {
+
+            String proxyAuthorization = httpRequest.getHeader(PROXY_AUTHORIZATION_HEADER);
+            if (proxyAuthorization == null) {
+                super.handle(pathInContext, request, httpRequest, httpResponse);
+            } else if (proxyAuthorization.equals(NTLM_MSG_TYPE_1)) {
```

Comment:
Add something to then assert that no header was received in the handler.

Reply:
Done. The assert verifies that the proxy authorization header must be null in this test case.